src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -194,15 +208,55 @@ private string GetResourceId()
             return this.GetPolicyArtifactFullyQualifiedId(this.Scope, Constants.MicrosoftAuthorizationPolicyAssignmentType, this.Name);
         }
 
+        /// <summary>
+        /// Verifies if identity type and/or identity ID and/or location are present
+        /// </summary>
+        private bool CheckIfIdentityPresent()
+        {
+            if (this.IdentityType != null & this.Location == null || this.AssignIdentity != null && this.AssignIdentity.IsPresent && this.Location == null)
+            {
+                throw new PSInvalidOperationException("Location needs to be specified if a managed identity is to be assigned to the policy assignment.");
+            }
+
+            if (this.IdentityType == ManagedIdentityType.UserAssigned && this.IdentityId == null)
+            {
+                throw new PSInvalidOperationException("A user assigned identity id needs to be specified if the identity type is 'UserAssigned'.");
+            }
+
+            if (this.IdentityType == ManagedIdentityType.SystemAssigned && this.IdentityId != null)
+            {
+                throw new PSInvalidOperationException("Cannot specify an identity ID if identity type is 'SystemAssigned'.");
+            }
+
+            return true;
+        }
+
         /// <summary>
         /// Constructs the resource
         /// </summary>
         private JToken GetResource()
         {
+            if (this.AssignIdentity != null && this.AssignIdentity.IsPresent)
+            {
+                this.IdentityType = ManagedIdentityType.SystemAssigned;
+            }
+
+            ResourceIdentity identityObject = this.IdentityType != null ?
+                (this.IdentityType == ManagedIdentityType.SystemAssigned ?
+                new ResourceIdentity { Type = ManagedIdentityType.SystemAssigned.ToString() } :
+                new ResourceIdentity
```

Comment:
Do you need to check this.IdentityType == ManagedIdentityType.UserAssigned? it can be ManagedIdentityType.None, right?

Reply:
You're right. Fixing.

Reply:
Hmmm, when this.IdentityType == ManagedIdentityType.None. it will create new ResourceIdentity { Type = ManagedIdentityType.None } in the new logic. 
Maybe change the check on the line 244 from this.IdentityType != null to this.IdentityType != null && this.IdentityType != ManagedIdentityType.None

Reply:
Talked offline. it should create ResourceIdentity { Type = ManagedIdentityType.None } for the case.